core/include/jml/model.hpp

```diff
@@ -10,10 +10,38 @@
 
 namespace jml {
 
+class Model_Layer {
+  friend class Layer_Actv_Funcs_List;
+
+private:
+  Matrix* layer_matrix;
+  Vector* layer_bias_vector;
+  // Array of pointers to function pointers (because lonely function pointers might not be contiguous in memory)
+  // The size of this will be dictated by the associated matrix
+  double (*layer_actv_funcs_arr[])();
```

Comment:
What do we need an entire array of function pointers for? Each layer should only have one activation function.

Reply:
For mixing/matching, and directly including function pointers might mean more memory space but faster accesses than enum -> function pointer lookups

Reply:
Explain what you mean by mixing/matching.

Reply:
Each layer has multiple perceptrons - with mixing and matching, the activation functions within each layer need not be identical which could produce results of experimental interest

[](https://discord.com/channels/909325954638299136/1153015102568677417/1156740693407039528)

[](https://discord.com/channels/909325954638299136/1153015102568677417/1157907596745904199)

Reply:
It is possible that we could keep it as one activation function for now and possibly experiment with that later / for other layer types?

Reply:
Jake's abstract class ActivationFunction in activation_functions.hpp will permit a very easy, modular implementation of this

Reply:
Okay, then keeping it sounds good. But do add the implementations of the class itself and nt just the header, and we should be good to go.